src/connectors/gcp/storage.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020-2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use crate::errors::Result;
+use reqwest::Client;
+use tremor_value::Value;
+
+pub(crate) async fn get_object(
+    client: &Client,
+    bucket_name: &str,
+    object_name: &str,
+) -> Result<Value<'static>> {
+    let url = format!(
```

Comment:
It would be nicer to use `Url` and build the URL instead of a string format. including variables like that opens this up to possible injection attacks as no escaping takes place. since the gcs code handles secrets we might want to be extra careful here.

(same below for the other `format!` calls)